### ISSUE TITLE
[Overview] Models differences 

### ISSUE THREAD

**Issue by YukiSakuma** [nagadomi/waifu2x#92]
I want to know the differences between the Ukbench, photo, RGB, and Y-model and what do you think is the best model in your opinion thanks..

**Comment by nagadomi** [comment]
Here is the results of internal benchmark.
TL;DR: For photo, `models/photo` is best. For artwork, `models/anime_style_art_rgb`  is best.

## dataset

    photo_test: 300 various photos.
    art_test  : 90 artworks (PNG only).

## 2x upscaling model

| Dataset/Model | anime\_style\_art(Y) | anime\_style\_art\_rgb | photo   | ukbench|
|---------------|----------------------|------------------------|---------|--------|
| photo\_test   |                29.83 |                  29.81 |**29.89**|  29.86 |
| art\_test     |                36.02 |               **36.24**|  34.92  |  34.85 |

The evaluation metric is PSNR(Y only), higher is better.

## Denosing level 1 model

| Dataset/Model            | anime\_style\_art | anime\_style\_art\_rgb | photo   |
|--------------------------|-------------------|------------------------|---------|
| photo\_test Quality 80   |             36.07 |               **36.20**|   36.01 |
| photo\_test Quality 50,45|             31.72 |                 32.01  |**32.31**|
| art\_test Quality 80     |             40.39 |               **42.48**|   40.35 |
| art\_test Quality 50,45  |             35.45 |               **36.70**|   36.27 |

The evaluation metric is PSNR(RGB), higher is better.

## Denosing level 2 model

| Dataset/Model            | anime\_style\_art | anime\_style\_art\_rgb | photo   |
|--------------------------|-------------------|------------------------|---------|
| photo\_test Quality 80   |             34.03 |                  34.42 |**36.06**|
| photo\_test Quality 50,45|             31.95 |                  32.31 |**32.42**|
| art\_test Quality 80     |             39.20 |               **41.12**|   40.48 |
| art\_test Quality 50,45  |             36.14 |               **37.78**|   36.55 |

The evaluation metric is PSNR(RGB), higher is better.

**Comment by nagadomi** [comment]
I am going to release new `models/anime_style_art`(Y only) model. It might be able to beat the RGB model.

**Comment by YukiSakuma** [comment]
Thanks what does it mean when using photo model? Can I use photo model for upscaling anime (video) screenshots?

**Comment by nagadomi** [comment]
Yes.
`anime_style_art_rgb` is specialized for digital illustration. Anime is made by video recording.
`anime_style_art_rgb` sometimes makes weird contour line for Anime(includes video frames and screenshots).

source:
![anime](https://cloud.githubusercontent.com/assets/287255/13896863/0d359fa4-ede0-11e5-8129-594385389c31.png)
anime_style_art_rgb:
![o1](https://cloud.githubusercontent.com/assets/287255/13896864/1b2374a6-ede0-11e5-84e6-14c5042f0da9.png)
photo:
![o2](https://cloud.githubusercontent.com/assets/287255/13896865/26aa9d86-ede0-11e5-97c8-5a256a16e949.png)

**Comment by YukiSakuma** [comment]
But I think in terms of noise reduction RGB is better than Photo for denoising video frames (anime)

**Comment by Kohit** [comment]
Hi, I'd like to know the differences between vgg_7 and upconv_7 in pre-trained models?

**Comment by nagadomi** [comment]
vgg_7 and upconv_7  have different network architectures.
vgg_7: 1. 2x the input image with nearest neighbor upscaler. 2. Repair the image with CNN
upconv_7: 1. end-to-end 2x with CNN (it uses deconvolution layer).

upconv_7 is 2~3x faster than vgg_7, and its upscaling quality is better than vgg_7.
Simply, upconv_7 is the new version, vgg_7 is the older version. But upconv_7 has [checkboard artifact ](http://distill.pub/2016/deconv-checkerboard/) problem. It happens very seldom, though.

**Comment by Kohit** [comment]
Thanks. I've read the  ESPCN paper. Is upconv_7's last layer like that (sub-pixel convolution)?

**Comment by nagadomi** [comment]
The purpose is the same, but the algorithm is different.
I tried subpixel convolution before the ESPCN paper was published([github log](https://github.com/nagadomi/waifu2x/blob/v0.12/lib/DepthExpand2x.lua)) but eventually I chose [deconvolution](https://github.com/torch/nn/blob/master/doc/convolution.md#spatialfullconvolution).

**Comment by splinter21** [comment]
> The purpose is the same, but the algorithm is different. I tried subpixel convolution before the ESPCN paper was published([github log](https://github.com/nagadomi/waifu2x/blob/v0.12/lib/DepthExpand2x.lua)) but eventually I chose [deconvolution](https://github.com/torch/nn/blob/master/doc/convolution.md#spatialfullconvolution).

Why did you choose deconvolution eventually? Is there any cases for comparision?
